app/src/main/java/com/example/dogbook/ComposeActivity/ComposeActivity.java

```diff
@@ -160,6 +198,22 @@ public void done(ParseException e) {
         });
     }
 
+    @NeedsPermission({Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION})
+    private void getCurrentLocation() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.ACCESS_FINE_LOCATION}, 1);
+        }
+        locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0, locationListener);
+
+    }
+
+    @Override
+    public void onLocationChanged(@NonNull Location location) {
+
+
+    }
```

Comment:
Do you have to implement this?  If so, would be good to say that it's a no-op. Otherwise, I think you can replace the location listener you have above with just updating in this method directly?